### PR TITLE
[GALL-1627] Partner loader query fix for partner analytics

### PR DESCRIPTION
**Pull request by sepans**
Fixes [GALL-1627](https://artsyproduct.atlassian.net/browse/GALL-1627)

## Problem:
`parseFieldASTsIntoArray` function assumes the top level `fieldASTs`s are of type `Field` so with `map(flatMap(fieldASTs, "selectionSet.selections"), "name.value")` it gets the name of all of them, but when that assumption is not true (e.g. top level `fieldAST` is an inline fragment) it doesn't go deeper into the tree.

This hasn't been previously a problem but when we stitched `analytics` under `partner` the top level AST is [this](https://github.com/artsy/metaphysics/blob/master/src/lib/stitching/vortex/stitching.ts#L239-L241) inline fragment and `parseFieldASTsIntoArray` returns `[undefined]` and the `queriedForFieldsOtherThanBlacklisted` function fails to recognize that there are no other fields in the query and always queries gravity for partner data even when the `id` is already provided in the query and no other data is needed from gravity.

## Solution
In the [first attempt](https://github.com/artsy/metaphysics/pull/1915) using a library but @alloy pointed out that we already have the exact same helper function available ❤️ 

So now using `includesFieldsOtherThanSelectionSet` instead of `parseFieldASTsIntoArray` in `queriedForFieldsOtherThanBlacklisted`

Finally I updated the specs to test the `partnerLoader` is only being called when the `id` passed to the query is the slug. 

It also required updating a bunch of loader classes to pass `GraphQLResolveInfo` instead of just `FieldNodes` to `queriedForFieldsOtherThanBlacklisted`

Other than tests passing also verified the result of `includesFieldsOtherThanSelectionSet` is exactly the same as the previous implementation when running tests.

## TODO:
- [ ] V2. (Just updating `analytics` to `_id` in `v2/partner.ts` is it still called `_id`?)

### CODE REVIEW

File: src/schema/v1/partner.ts [hunk 1]
```diff
@@ -290,14 +290,15 @@ const Partner: GraphQLFieldConfig<void, ResolverContext> = {
       description: "The slug or ID of the Partner",
     },
   },
-  resolve: (_root, { id }, { partnerLoader }, { fieldNodes }) => {
-    const blacklistedFields = ["analytics"]
+  resolve: (_root, { id }, { partnerLoader }, info) => {
+    // when the query is partner analytics, vortex stitching fragment only has _id
+    const blacklistedFields = ["_id"]
```
Comment (sepans) [inline src/schema/v1/partner.ts]: I originally thought we had to blacklist `analytics` for queries like this:
```graphql
query {
  partner(id: "4d8b92c44eb68a1b2c0004cb") {
    analytics {
      topShows: rankedStats(first: 1, objectType: SHOW, period: ONE_YEAR) {
         ...
      }
    }
  }
}
```

but realized when the analytics query gets to this point the query is the [stitching fragment](https://github.com/artsy/metaphysics/blob/master/src/lib/stitching/vortex/stitching.ts#L239-L241) so need to blacklist (whitelist) `_id`

File: src/lib/helpers.ts [hunk 2]
```diff
@@ -85,20 +84,13 @@ export const stripTags = (str?: string) => {
...
   blacklistedFields
 ) => {
-  if (!fieldASTs) return true
-  const queriedFields = parseFieldASTsIntoArray(fieldASTs)
-  return difference(queriedFields, blacklistedFields).length > 0
-}
-export const queryContainsField = (fieldASTs, soughtField) => {
-  return parseFieldASTsIntoArray(fieldASTs).includes(soughtField)
+  if (!info) return true
+  return includesFieldsOtherThanSelectionSet(info, blacklistedFields)
```
Comment (oxaudo) [inline src/lib/helpers.ts]: Did not have time to go into too much details but stealing @joeyAghion usual comment here. I think we had a discussion how `blacklisted` is not a preferred terminology now. Maybe something with `deny` is a better option?

**Comment by sepans** [comment]
@alloy I am a bit confused about `v2`. As I understood it that v1 `_id` comes from [this](https://github.com/artsy/metaphysics/blob/master/src/lib/stitching/vortex/stitching.ts#L239-L241) fragment. Does it go through the schema transformation before getting to the loader in v2, or we need to also update the stitching code for v2? (well not that either of those are required now but wanted to keep that in mind for future)

**Comment by alloy** [comment]
@sepans Hmm, this is an excellent point, thanks for raising it. I think we’ll notice soon enough when we migrate reaction 😬 /cc @zephraph

**Comment by alloy** [comment]
@sepans Sorry, I lost track of this. Is it ready for merging?

**Comment by sepans** [comment]
I believe so. I addressed the comments and the [v2 problem](https://github.com/artsy/metaphysics/pull/1918#issuecomment-526604387) doesn't seem to be something we need to worry about now.

**Comment by alloy** [comment]
Great, thanks for bearing with me! 🙌

**Pull request merged**